Validation/MuonHits/src/CSCSimHitMatcher.cc

```diff
@@ -0,0 +1,382 @@
+#include "Validation/MuonHits/interface/CSCSimHitMatcher.h"
+
+using namespace std;
+
+CSCSimHitMatcher::CSCSimHitMatcher(const edm::ParameterSet& ps,
+                                   edm::ConsumesCollector&& iC)
+    : MuonSimHitMatcher(ps, std::move(iC)) {
+  simHitPSet_ = ps.getParameterSet("cscSimHit");
+  verbose_ = simHitPSet_.getParameter<int>("verbose");
+  simMuOnly_ = simHitPSet_.getParameter<bool>("simMuOnly");
+  discardEleHits_ = simHitPSet_.getParameter<bool>("discardEleHits");
+
+  simHitInput_ = iC.consumes<edm::PSimHitContainer>(
+      simHitPSet_.getParameter<edm::InputTag>("inputTag"));
+}
+
+/// initialize the event
+void CSCSimHitMatcher::init(const edm::Event& iEvent,
+                            const edm::EventSetup& iSetup) {
+  iSetup.get<MuonGeometryRecord>().get(csc_geom_);
+  if (csc_geom_.isValid()) {
+    geometry_ = &*csc_geom_;
+  } else {
+    hasGeometry_ = false;
+    std::cout << "+++ Info: CSC geometry is unavailable. +++\n";
```

Comment:
@dildick please use here and in all other places the MessageLogger, cout is not thread safe 9even if conditional to a verbose flag), and set an appropriate verbosity level (in this case most probably a LogWarning is suitable).